src/sentry/static/sentry/app/views/eventsV2/landing.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import DocumentTitle from 'react-document-title';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import * as ReactRouter from 'react-router';
+import {Params} from 'react-router/lib/Router';
+import {Location} from 'history';
+
+import {t} from 'app/locale';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import SentryTypes from 'app/sentryTypes';
+import {Organization} from 'app/types';
+import localStorage from 'app/utils/localStorage';
+import withOrganization from 'app/utils/withOrganization';
+
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import Banner from 'app/components/banner';
+import Button from 'app/components/button';
+import Feature from 'app/components/acl/feature';
+import SearchBar from 'app/views/events/searchBar';
+import NoProjectMessage from 'app/components/noProjectMessage';
+
+import {PageContent, PageHeader} from 'app/styles/organization';
+import space from 'app/styles/space';
+
+import Events from './events';
+import SavedQueryButtonGroup from './savedQuery';
+import EventView from './eventView';
+import EventInputName from './eventInputName';
+import {DEFAULT_EVENT_VIEW_V1} from './data';
+import QueryList from './queryList';
+import DiscoverBreadcrumb from './breadcrumb';
+
+const DISPLAY_SEARCH_BAR_FLAG = false;
+
+type Props = {
+  organization: Organization;
+  location: Location;
+  router: ReactRouter.InjectedRouter;
+  params: Params;
+};
+
+class DiscoverLanding extends React.Component<Props> {
+  static propTypes: any = {
+    organization: SentryTypes.Organization.isRequired,
+    location: PropTypes.object.isRequired,
+    router: PropTypes.object.isRequired,
+  };
+
+  state = {
+    isBannerHidden: localStorage.getItem('discover-banner-dismissed'),
+  };
+
+  getDocumentTitle = (name: string | undefined): Array<string> => {
+    return typeof name === 'string' && String(name).trim().length > 0
+      ? [String(name).trim(), t('Discover')]
+      : [t('Discover')];
+  };
+
+  handleClick = () => {
+    localStorage.setItem('discover-banner-dismissed', true);
+    this.setState({isBannerHidden: true});
+  };
+
+  renderBanner() {
+    const bannerDismissed = localStorage.getItem('discover-banner-dismissed');
```

Comment:
Should this read from state? Right now the state and localstorage could get out of sync.

Reply:
should be addressed in https://github.com/getsentry/sentry/pull/15562/commits/957651781fc3e617fdbe03526fea204158bb21da